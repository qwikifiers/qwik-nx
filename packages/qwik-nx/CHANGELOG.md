# Changelog

This file was generated using [@jscutlery/semver](https://github.com/jscutlery/semver).

## [0.2.1](https://github.com/qwikifiers/qwik-nx/compare/qwik-nx-0.2.0...qwik-nx-0.2.1) (2022-12-07)


### Bug Fixes

* fix npm publish dist ([758ca4a](https://github.com/qwikifiers/qwik-nx/commit/758ca4ad1b21c504553f502fcdafd03152b55408))



# [0.2.0](https://github.com/qwikifiers/qwik-nx/compare/qwik-nx-0.1.0...qwik-nx-0.2.0) (2022-12-07)


### Features

* application & library generator updates ([#5](https://github.com/qwikifiers/qwik-nx/issues/5)) ([5c6b7b9](https://github.com/qwikifiers/qwik-nx/commit/5c6b7b92a5f5525baa0a3467f6b31a19ff155af9))



# 0.1.0 (2022-12-05)


### Bug Fixes

* add missing publish scripts ([5b63887](https://github.com/qwikifiers/qwik-nx/commit/5b638879c5462122c982b64d873af8d97e4819a3))


### Features

* library generator ([ae8a17b](https://github.com/qwikifiers/qwik-nx/commit/ae8a17babe16db6fea60c40e9e98ec1bfd4e2411))
