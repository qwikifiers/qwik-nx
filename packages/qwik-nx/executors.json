{
  "$schema": "http://json-schema.org/schema",
  "executors": {
    "build": {
      "implementation": "./src/executors/build/executor",
      "schema": "./src/executors/build/schema.json",
      "description": "Build a Qwik application."
    },
    "micro-frontends-dev-server": {
      "implementation": "./src/executors/micro-frontends-dev-server/executor",
      "schema": "./src/executors/micro-frontends-dev-server/schema.json",
      "description": "Serve a host application along with its known remotes."
    },
    "micro-frontends-preview-server": {
      "implementation": "./src/executors/micro-frontends-preview-server/executor",
      "schema": "./src/executors/micro-frontends-preview-server/schema.json",
      "description": "Serve a host application along with its known remotes in a preview mode."
    }
  }
}
