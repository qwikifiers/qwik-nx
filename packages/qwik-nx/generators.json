{
  "$schema": "http://json-schema.org/schema",
  "name": "qwik-nx",
  "version": "0.0.1",
  "generators": {
    "application": {
      "factory": "./src/generators/application/generator",
      "schema": "./src/generators/application/schema.json",
      "description": "Qwik application generator",
      "aliases": ["app", "a"]
    },
    "init": {
      "factory": "./src/generators/init/init",
      "schema": "./src/generators/init/schema.json",
      "description": "Initialize the qwik-nx plugin",
      "aliases": ["i"]
    },
    "route": {
      "factory": "./src/generators/route/generator",
      "schema": "./src/generators/route/schema.json",
      "description": "Adding a Qwik City route",
      "aliases": ["r"]
    },
    "component": {
      "factory": "./src/generators/component/generator",
      "schema": "./src/generators/component/schema.json",
      "description": "Generating a Qwik component",
      "aliases": ["cmp", "c"]
    },
    "library": {
      "factory": "./src/generators/library/generator",
      "schema": "./src/generators/library/schema.json",
      "description": "Create a Qwik Library",
      "aliases": ["lib", "l"]
    },
    "preset": {
      "factory": "./src/generators/preset/generator",
      "schema": "./src/generators/preset/schema.json",
      "description": "Create a Qwik Workspace Preset",
      "hidden": true
    },
    "setup-tailwind": {
      "factory": "./src/generators/setup-tailwind/setup-tailwind",
      "schema": "./src/generators/setup-tailwind/schema.json",
      "description": "Set up Tailwind configuration for a project.",
      "aliases": ["tailwind", "tw", "t"]
    }
  }
}
