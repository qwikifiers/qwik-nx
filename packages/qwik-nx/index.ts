// This is explicitly empty, but serves as a primary compilation entry-point.
export default {};
