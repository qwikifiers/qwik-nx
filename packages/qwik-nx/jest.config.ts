/* eslint-disable */
export default {
  displayName: 'qwik-nx',
  preset: '../../jest.preset.js',
  globals: {},
  transform: {
    '^.+\\.[tj]s$': [
      'ts-jest',
      {
        tsconfig: '<rootDir>/tsconfig.spec.json',
      },
    ],
  },
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../coverage/packages/qwik-nx',
};
