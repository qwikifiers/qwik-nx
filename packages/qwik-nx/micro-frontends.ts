export * from './src/micro-frontends';
