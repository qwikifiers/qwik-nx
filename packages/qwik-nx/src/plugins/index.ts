export * from './qwik-nx-vite.plugin';
export type {
  ProjectFilter,
  QwikNxVitePluginOptions,
} from './models/qwik-nx-vite';
