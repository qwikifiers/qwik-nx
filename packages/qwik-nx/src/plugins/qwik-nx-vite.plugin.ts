import { type Plugin } from 'vite';
import {
  QwikNxVitePluginOptions,
  QwikVitePluginStub,
} from './models/qwik-nx-vite';
import { getVendorRoots } from './utils/get-vendor-roots';

/**
 * `qwikNxVite` plugin serves as an integration step between Qwik and Nx.
 * At this point its main purpose is to provide Nx libraries as vendor roots for the Qwik.
 * This is required in order for the optimizer to be able to work with entities imported from those libs.
 *
 * By default `qwikNxVite` plugin will provide Qwik with paths of Nx projects,
 * that are recognized as dependencies of the current one and are specified in the tsconfig.base.json.
 * this behavior is customizable: you can exclude additional projects by providing `options.excludeProjects.keepUnrelatedProjects`,
 * or completely control what's included by using `options.includeProjects`
 */
export function qwikNxVite(options?: QwikNxVitePluginOptions): Plugin {
  const vitePlugin: Plugin = {
    name: 'vite-plugin-qwik-nx',
    enforce: 'pre',

    async configResolved(config) {
      const qwikPlugin = config.plugins.find(
        (p) => p.name === 'vite-plugin-qwik'
      ) as QwikVitePluginStub;
      if (!qwikPlugin) {
        throw new Error('Missing vite-plugin-qwik');
      }

      const pluginOptions = qwikPlugin.api.getOptions();

      const vendorRoots = await getVendorRoots(options, pluginOptions);

      pluginOptions.vendorRoots.push(...vendorRoots);
    },
  };

  return vitePlugin;
}
