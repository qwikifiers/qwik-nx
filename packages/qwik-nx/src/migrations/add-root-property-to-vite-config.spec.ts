import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';

import appGenerator from '../generators/application/generator';
import update from './add-root-property-to-vite-config';

describe('add-root-property-to-vite-config migration', () => {
  let tree: Tree;

  beforeEach(async () => {
    tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
    await appGenerator(tree, { directory: 'apps/myapp' });
  });

  it('should add/update the "root" property', async () => {
    const viteConfigPath = `apps/myapp/vite.config.ts`;
    // just to ensure the file at that paths exists before updating it
    expect(tree.exists(viteConfigPath)).toBeTruthy();

    tree.write(viteConfigPath, getViteConfigSample());
    update(tree);
    expect(tree.read(viteConfigPath, 'utf-8')).toMatchSnapshot();
  });
});

const getViteConfigSample = (
  includeRootProp = false
) => `import { qwikVite } from '@builder.io/qwik/optimizer';
import { defineConfig } from 'vite';
import tsconfigPaths from 'vite-tsconfig-paths';
import { qwikNxVite } from 'qwik-nx/plugins';

export default defineConfig({
  ${includeRootProp ? '\nroot: "apps/myapp"' : ''}
  plugins: [
    qwikNxVite(),
    qwikVite({
      client: {
        outDir: '../../dist/apps/myapp/client',
      },
      ssr: {
        outDir: '../../dist/apps/myapp/server',
      },
    }),
    tsconfigPaths({ root: '../../' }),
  ],
});
`;
