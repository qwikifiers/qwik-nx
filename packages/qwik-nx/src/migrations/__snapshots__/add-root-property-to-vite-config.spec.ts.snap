// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add-root-property-to-vite-config migration should add/update the "root" property 1`] = `
"import { qwikVite } from \\"@builder.io/qwik/optimizer\\";
import { defineConfig } from \\"vite\\";
import tsconfigPaths from \\"vite-tsconfig-paths\\";
import { qwikNxVite } from \\"qwik-nx/plugins\\";
export default defineConfig({
    root: apps / myapp,
    plugins: [
        qwikNxVite(),
        qwikVite({
            client: {
                outDir: \\"../../dist/apps/myapp/client\\",
            },
            ssr: {
                outDir: \\"../../dist/apps/myapp/server\\",
            },
        }),
        tsconfigPaths({ root: \\"../../\\" }),
    ]
});
"
`;
